lib/app/data/repositories/user_repository.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class UserRepository {
+  final FirebaseFirestore firestore;
+
+  UserRepository(this.firestore);
+
+  Future<void> saveUserData({
+    required String userId,
+    required String username,
+    required String email,
+    // Add other user data fields as needed
+  }) async {
+    try {
+      await firestore.collection('users').doc(userId).set({
```

Comment:
Why using firestore when you have access to Supabase in the whole project @didierganthier

Reply:
Oh my bad 🤣 I've been working on so many Firebase projects I got confused here. My bad. Fixing this asap.